src/main/flight/pid.c

```diff
@@ -1343,7 +1355,11 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
         // b = 1 and only c (feedforward weight) can be tuned (amount derivative on measurement or error).
 
         // -----calculate P component
-        pidData[axis].P = pidCoefficient[axis].Kp * errorRate * tpaFactorKp;
+        // If enabled vary P and D using the content of AUXn (Heli-like vehicles, gyro-sensitivity channel).
+        if (pidAdjustmentChannel) {
+            auxPgain = ((-0.1f * constrainf(((float)rcData[pidAdjustmentChannel]), 1000.0f, 2000.0f)) + 200.0f) / 50.0f;
```

Comment:
Please see `scalerangef`. 
something like `scalerangef(constain(rcData[pidAdjustmentChannel], PWM_RANGE_MIN, PWM_RANGE_MAX), PWM_RANGE_MIN, PWM_RANGE_MAX, <ming gain>, <max gain>));` should do, probably even without constrain

Also note that with current values gain can be set to 0, completely disabling pid